tamr_unify_client/dataset/resource.py

```diff
@@ -434,3 +444,88 @@ def _geo_attr_names():
             "polygon",
             "multiPolygon",
         }
+
+
+class DatasetSpec:
+    """A representation of the server view of a dataset."""
+
+    def __init__(self, client, data, api_path):
```

Comment:
nit: I suggest we use `client, api_path, data` ordering to group together the things that won't change (i.e. `client` and `api_path`) together, and keep `data` at the end.

even more minor nit: Additionally, you could consider ordering the assigments within the `__init__` method in the same order as they are declared in the `__init__` function signature.

Reply:
We should use the same order as `Dataset.__init__()`, otherwise users will get it wrong every single time.

Reply:
@nbateshaus users only need to use `.spec()` so they never have to order it the same.